src/content/concepts/module-federation.md

```diff
@@ -0,0 +1,95 @@
+
+This is often known as Micro-Frontends, but is not limited to that.
+
+## Low-level concepts
+
+We distinguish between local and remote modules. Local modules are normal modules which are part of the current build. Remote modules are modules that are not part of the current build and loaded from a so-called container at the runtime.
+
+Loading remote modules is considered async operation. When using a remote module these async operations will be placed in the next chunk loading operation(s) that is between the remote module and the entrypoint. It's not possible to use a remote module without a chunk loading operation.
+
+A chunk loading operation is usually an `import()` call, but older constructs like `require.ensure` or `require([...])` are supported as well.
+
+A container is created through a container entry, which exposes asynchronous access to the specific modules. The exposed access is separated into two steps:
+
+1. loading the module (asynchronous)
+2. evaluating the module (synchronous).
+
+Step 1 will be done during the chunk loading. Step 2 will be done during the module evaluation interleaved with other (local and remote) modules. This way, evaluation order is unaffected by converting a module from local to remote or the other way around.
+
+It is possible to nest a container. Containers can use modules from other containers. Circular dependencies between container are also possible.
+
+### Overriding
+
+A container is able to flag selected local modules as "overridable". A consumer of the container is able to provide "overrides", which are modules that replace one of the overridable modules of the container. All modules of the container will use the replacement module instead of the local module when the consumer provides one. When the consumer doesn't provide a replacement module, all modules of the container will use the local one.
+
+The container will manage overridable modules in a way that they do not need to be downloaded when they have been overridden by the consumer. This usually happens by placing them into separate chunks.
+
+On the other hand, the provider of the replacement modules, will only provide asynchronous loading functions. It allows the container to load replacement modules only when they are needed. The provider will manage replacement modules in a way that they do not need to be downloaded at all when they are not requested by the container. This usually happens by placing them into separate chunks.
+
+A "name" is used to identify overridable modules from the container.
+
+Overrides are provided in a similar way as the container exposes modules, separated into two steps:
+
+1. Loading (asynchronous)
+2. evaluating (asynchronous)
+
+W> When nesting is used, providing overrides to one container will automatically override the modules with the same "name" in the nested container(s).
+
+Overrides must be provided before the modules of the container are loaded, otherwise behavior is undefined.
```

Comment:
Maybe we can ask sokra for help about this `undefined` thing here? Or just remove `otherwise behavior is undefined`.

Reply:
yes lets remove for now